src/helpers/mathjax.coffee

```diff
@@ -40,10 +41,26 @@ typesetMath = (root) ->
   # submit all nodes at once for optimal rendering performance
   window.MathJax.Hub.Queue(['Typeset', MathJax.Hub, _.toArray(nodes)])
 
+  cleanMathArtifacts()
+
+# Install a debounce around typesetting function so that it will only run once
+# every 10ms even if called multiple calls times in that period
+TypesetDocument = _.debounce( TypesetDocument, 10)
+
+
+typesetMath = (root) ->
+  nodes = root.querySelectorAll(MATH_SELECTOR) or []
+  hasMath = root.querySelector('math')
```

Comment:
Should this be `root.querySelector("math, #{MATH_SELECTOR}")` to handle math in the books (`math`) as well as math in the exercises (`MATH_SELECTOR`)?